src/sinol_make/commands/run/__init__.py

```diff
@@ -239,14 +236,11 @@ def execute_oiejq(self, command, input_file_path, output_file_path, answer_file_
 
     def execute_time(self, command, result_file_path, input_file_path, output_file_path, answer_file_path,
                      time_limit, memory_limit):
-        input_file = open(input_file_path, "r")
-        output_file = open(output_file_path, "w")
-        process = subprocess.Popen(command, stdin=input_file, stdout=output_file, stderr=subprocess.DEVNULL)
-        process.wait()
+        with open(output_file_path, "w") as output_file, open(input_file_path, "r") as input_file:
+            process = subprocess.Popen(command, stdin=input_file, stdout=output_file, stderr=subprocess.DEVNULL)
+            process.wait()
+            output_file.flush()
```

Comment:
Do we have to save the output to a file? I guess that Python handles capturing the output to a variable, without a new file. Using files with absolute paths makes it easier to make a bug by e.g. running two sinol-makes at the same time. In short, when possible, it's better to redirect directly to a variable or a path with a random name, rather than to a file. Especially that the file is not getting deleted and it uses memory in the filesystem.

Reply:
Also, the `flush` here might not be needed, right? I'm not sure, but the file might be flushed on closing.

Reply:
I changed so that `with` statements are now used